include/xtensor/xchunked_array.hpp

```diff
@@ -76,12 +101,49 @@ namespace xt
                 arr0[i] = std::get<0>(arr[i]);
                 arr1[i] = std::get<1>(arr[i]);
             }
-            return std::make_tuple(arr0, arr1);
+            return std::make_pair(arr0, arr1);
         }
 
-        xt::xarray<chunk_type> m_chunks;
-        std::vector<size_t> m_shape;
-        std::vector<size_t> m_chunk_shape;
+        template <class It>
+        inline std::pair<std::vector<size_t>, std::vector<size_t>> get_indexes_dynamic(It first, It last) const
+        {
+            std::vector<size_t> indexes_of_chunk;
+            std::vector<size_t> indexes_in_chunk;
+            std::pair<size_t, size_t> chunk_index;
+            int dim = 0;
```

Comment:
Since `dim` will be used to address containers, it should probably be size_t or size_type